requests/models.py

```diff
@@ -403,10 +404,15 @@ def prepare_url(self, url, params):
     def prepare_headers(self, headers):
         """Prepares the given HTTP headers."""
 
+        self.headers = CaseInsensitiveDict()
         if headers:
-            self.headers = CaseInsensitiveDict((to_native_string(name), value) for name, value in headers.items())
-        else:
-            self.headers = CaseInsensitiveDict()
+            #Regex checks that string doesn't begin with whitespace or contain return characters.
+            #Otherwise, it checks if it's a blank string.
+            pat = re.compile("^\S[^\r\n]*$|^$")
```

Comment:
We need to bring `to_native_string` up to before the match on this regex. This is because we allow users to provide headers as unicode and byte strings, which will misbehave with a native string regex.